gemrb/core/Map.cpp

```diff
@@ -2793,246 +2796,203 @@ PathNode* Map::GetLine(const Point &start, const Point &dest, int Speed, int Ori
 	return Return;
 }
 
-/*
- * find a path from start to goal, ending at the specified distance from the
- * target (the goal must be in sight of the end, if 'sight' is specified)
- *
- * if you don't need to find an optimal path near the goal then use FindPath
- * instead, but don't change this one without testing with combat and dialog,
- * you can't predict the goal point for those, you *must* path!
- */
-PathNode* Map::FindPathNear(const Point &s, const Point &d, unsigned int size, unsigned int MinDistance, bool sight)
-{
-	// adjust the start/goal points to be searchmap locations
-	Point start( s.x/16, s.y/12 );
-	Point goal ( d.x/16, d.y/12 );
-	Point orig_goal = goal;
+typedef Point NavmapPoint;
+typedef Point SearchmapPoint;
 
-	// re-initialise the path finding structures
-	memset( MapSet, 0, Width * Height * sizeof( unsigned short ) );
-	while (InternalStack.size())
-		InternalStack.pop();
+class PQNode {
+public:
+	PQNode(Point p, unsigned int l):first(p), second(l) {};
+	PQNode():first(Point(0, 0)), second(0) {};
 
-	// set the start point in the path finding structures
-	unsigned int pos2 = ( goal.x << 16 ) | goal.y;
-	unsigned int pos = ( start.x << 16 ) | start.y;
-	InternalStack.push( pos );
-	MapSet[start.y * Width + start.x] = 1;
+	Point first;
+	unsigned int second;
+
+	friend bool operator< (const PQNode &lhs, const PQNode &rhs) { return lhs.second < rhs.second;}
+	friend bool operator> (const PQNode& lhs, const PQNode& rhs){ return rhs < lhs; }
+	friend bool operator<= (const PQNode& lhs, const PQNode& rhs){ return !(lhs > rhs); }
+	friend bool operator>= (const PQNode& lhs, const PQNode& rhs){ return !(lhs < rhs); }
+	friend bool operator == (const PQNode& lhs, const PQNode& rhs) { return lhs.first == rhs.first; }
+	friend bool operator != (const PQNode& lhs, const PQNode& rhs) { return !(lhs == rhs); }
+
+};
 
-	unsigned int squaredmindistance = MinDistance * MinDistance;
-	bool found_path = false;
-	while (InternalStack.size()) {
-		pos = InternalStack.front();
-		InternalStack.pop();
-		unsigned int x = pos >> 16;
-		unsigned int y = pos & 0xffff;
 
-		if (pos == pos2) {
-			// we got all the way to the target!
-			found_path = true;
+/*
+ * find a path from start to goal, ending at the specified distance from the
+ * target (the goal must be in sight of the end, if 'sight' is specified)
+ */
+PathNode *Map::FindPath(const Point &s, const Point &d, unsigned int size,
+						unsigned int minDistance, bool sight) {
+	static const char *logString = "PathFinderWIP";
+	static const unsigned short MAX_PATH_COST = std::numeric_limits<unsigned short>::max();
+	static const size_t DEGREES_OF_FREEDOM = 8;
+	static const char dx[DEGREES_OF_FREEDOM] = {1, 0, -1, 0, 1, 1, -1, -1};
+	static const char dy[DEGREES_OF_FREEDOM] = {0, 1, 0, -1, 1, -1, 1, -1};
+	static const float weight = 1.5;	// Weighted heuristic. Finds sub-optimal paths
+										// but should be quite a bit faster
```

Comment:
If you put the whole comment on its own line, there'll be no need to break it in half. It looks odd for higher spaces-per-tab count views like here on github.